jenkins-job-components/src/main/java/com/ofg/job/component/JobComponent.java

```diff
@@ -0,0 +1,20 @@
+package com.ofg.job.component;
+
+import javaposse.jobdsl.dsl.Job;
+
+/**
+ * Interface for configurations applicable on Job.
+ */
+public interface JobComponent<J extends Job> {
+
+    JobComponent<? extends Job> EMPTY = job -> {};
+    
+    default <NJ extends J> JobComponent<NJ> and(JobComponent<NJ> other) {
```

Comment:
Ok so this idea has one flaw, that I think is not worth adding such methods. I've precisely added only one method, that can either specify more the job or stay at the current level of abstraction, so you can do:
```
AbstractJobConfigurer() &
AbstractJobConfigurer() &
AbstractJobConfigurer() &
AbstractJobConfigurer() &
```
or you can do 

```
AbstractJobConfigurer() &
AbstractJobConfigurer() &
AbstractJobConfigurer() &
MultiJobConfigurer() &
MultiJobConfigurer() &
MultiJobConfigurer() &
```
but you can't do something like:
```
AbstractJobConfigurer() &
MultiJobConfigurer() &
AbstractJobConfigurer() &
MavenJobConfigurer() &
```

and you can't do the later since it's silly (it actually won't compile with @CompileStatic) and since it would blow up with ClassCastException, the moment you would do
```
(
    AbstractJobConfigurer() &
    MultiJobConfigurer() &
    AbstractJobConfigurer() &
    MavenJobConfigurer() &
).applyOn job
```
That's the reason to restrict this to be able to move only one way.